### PR TITLE
Smoke spells readability

### PR DESCRIPTION
**Pull request by S34NW**
## What Does This PR Do
Tweaks the parts of code that define smoke types when casting spells, to make it more readable.
Uses defines for smoke types now, as opposed to random numbers not properly documented anywhere.

## Why It's Good For The Game
12 year old messy code that makes no sense to anyone looking at it is bad.

## Testing
I make my wizard sleep due to his own smoke

## Changelog
fix: Some mech smoke generator mechanics are now fixed

### CODE REVIEW

File: code/game/objects/effects/effect_system/effect_system.dm [hunk 1]
```diff
@@ -34,15 +34,15 @@ would spawn and follow the beaker, even if it is carried or thrown.
...
+/datum/effect_system/proc/set_up(amount = 3, only_cardinals = 0, source)
+	if(amount > 10)
+		amount = 10
+	number = amount
+	cardinals = only_cardinals
+	if(isturf(source))
+		location = source
 	else
-		location = get_turf(loca)
+		location = get_turf(source)
```
Comment (Contrabang) [inline code/game/objects/effects/effect_system/effect_system.dm:45]: Is there any reason why we're not just `get_turf(source)` for this check? wouldn't that do the same thing if source was a turf?

File: code/game/objects/effects/effect_system/effects_smoke.dm [hunk 2]
```diff
@@ -78,17 +78,17 @@
...
+		amount = 20
+	number = amount
+	cardinals = only_cardinals
+	if(isturf(source))
+		location = source
 	else
-		location = get_turf(loca)
-	if(direct)
-		direction = direct
+		location = get_turf(source)
```
Comment (Contrabang) [inline code/game/objects/effects/effect_system/effects_smoke.dm:89]: same question here, is this isturf check needed?

File: code/game/objects/effects/effect_system/effect_system.dm [hunk 3]
```diff
@@ -34,15 +34,12 @@ would spawn and follow the beaker, even if it is carried or thrown.
...
-/datum/effect_system/proc/set_up(n = 3, c = 0, loca)
-	if(n > 10)
-		n = 10
-	number = n
-	cardinals = c
-	if(isturf(loca))
-		location = loca
-	else
-		location = get_turf(loca)
+/datum/effect_system/proc/set_up(amount = 3, only_cardinals = 0, source)
```
Comment (Contrabang) [inline code/game/objects/effects/effect_system/effect_system.dm:37]:
```suggestion
/datum/effect_system/proc/set_up(amount = 3, only_cardinals = FALSE, source)
```

File: code/datums/spell.dm [hunk 4]
```diff
@@ -401,19 +402,17 @@ GLOBAL_LIST_INIT(spells, typesof(/obj/effect/proc_holder/spell))
...
+		if(smoke_type)
+			var/datum/effect_system/smoke_spread/smoke
+			switch(smoke_type)
+				if(SMOKE_HARMLESS)
+					smoke = new /datum/effect_system/smoke_spread()
+				if(SMOKE_COUGHING)
+					smoke = new /datum/effect_system/smoke_spread/bad()
+				if(SMOKE_SLEEPING)
+					smoke = new /datum/effect_system/smoke_spread/sleeping()
+			smoke.set_up(smoke_amt, 0, location)
```
Comment (Contrabang) [inline code/datums/spell.dm:414]:
```suggestion
			smoke.set_up(smoke_amt, FALSE, location)
```

File: code/game/objects/effects/effect_system/effect_system.dm [hunk 5]
```diff
@@ -34,15 +34,12 @@ would spawn and follow the beaker, even if it is carried or thrown.
...
+	if(amount > 10)
+		amount = 10
```
Comment (Contrabang) [inline code/game/objects/effects/effect_system/effect_system.dm:39]: Would this be better as a clamp? I'm not sure

File: code/game/objects/effects/effect_system/effects_smoke.dm [hunk 6]
```diff
@@ -176,7 +173,7 @@
 		for(var/obj/item/Item in T)
 			Item.extinguish()
 
-/datum/effect_system/smoke_spread/freezing/set_up(n = 5, c = 0, loca, direct, blasting = 0)
+/datum/effect_system/smoke_spread/freezing/set_up(amount = 5, only_cardinals = 0, source, desired_direction, blasting = 0)
```
Comment (Contrabang) [inline code/game/objects/effects/effect_system/effects_smoke.dm:176]: ditto about boolean only_cardinals

File: code/game/objects/items/weapons/grenades/smokebomb.dm [hunk 7]
```diff
@@ -19,7 +19,7 @@
 
 /obj/item/grenade/smokebomb/prime()
 	playsound(src.loc, 'sound/effects/smoke.ogg', 50, 1, -3)
```
Comment (Contrabang) [inline code/game/objects/items/weapons/grenades/smokebomb.dm:21]: out of scope maybe but...
```suggestion
	playsound(loc, 'sound/effects/smoke.ogg', 50, 1, -3)
```

File: code/game/objects/items/weapons/tanks/watertank.dm [hunk 8]
```diff
@@ -333,7 +333,7 @@
 
 /obj/effect/nanofrost_container/proc/Smoke()
 	var/datum/effect_system/smoke_spread/freezing/S = new
-	S.set_up(6, 0, loc, null, 1)
+	S.set_up(6, FALSE, loc, null, 1)
 	S.start()
 	var/obj/effect/decal/cleanable/flour/F = new /obj/effect/decal/cleanable/flour(src.loc)
```
Comment (Contrabang) [inline code/game/objects/items/weapons/tanks/watertank.dm:338]:
```suggestion
	var/obj/effect/decal/cleanable/flour/F = new /obj/effect/decal/cleanable/flour(loc)
```

File: code/modules/atmospherics/machinery/pipes/simple/pipe_simple.dm [hunk 9]
```diff
@@ -91,7 +91,7 @@
 	src.visible_message("<span class='danger'>\The [src] bursts!</span>")
 	playsound(src.loc, 'sound/effects/bang.ogg', 25, 1)
```
Comment (Contrabang) [inline code/modules/atmospherics/machinery/pipes/simple/pipe_simple.dm:92]:
```suggestion
	playsound(loc, 'sound/effects/bang.ogg', 25, 1)
```

File: code/modules/atmospherics/machinery/pipes/simple/pipe_simple.dm [hunk 10]
```diff
@@ -91,7 +91,7 @@
...
 	var/datum/effect_system/smoke_spread/smoke = new
-	smoke.set_up(1,0, src.loc, 0)
+	smoke.set_up(1, FALSE, src.loc)
```
Comment (Contrabang) [inline code/modules/atmospherics/machinery/pipes/simple/pipe_simple.dm:94]:
```suggestion
	smoke.set_up(1, FALSE, loc)
```

File: code/modules/mob/living/simple_animal/hostile/hivebot.dm [hunk 11]
```diff
@@ -75,7 +75,7 @@
 /mob/living/simple_animal/hostile/hivebot/tele/Initialize(mapload)
 	. = ..()
 	var/datum/effect_system/smoke_spread/smoke = new
-	smoke.set_up(5, 0, src.loc)
+	smoke.set_up(5, FALSE, src.loc)
 	smoke.start()
 	visible_message("<span class='danger'>[src] warps in!</span>")
 	playsound(src.loc, 'sound/effects/empulse.ogg', 25, 1)
```
Comment (Contrabang) [inline code/modules/mob/living/simple_animal/hostile/hivebot.dm:81]:
```suggestion
	smoke.set_up(5, FALSE, loc)
	smoke.start()
	visible_message("<span class='danger'>[src] warps in!</span>")
	playsound(loc, 'sound/effects/empulse.ogg', 25, 1)
```

File: code/modules/power/smes.dm [hunk 12]
```diff
@@ -425,7 +425,7 @@
 				M.show_message("<span class='warning'>[src] is making strange noises!</span>", 3, "<span class='warning'>You hear sizzling electronics.</span>", 2)
 			sleep(10*pick(4,5,6,7,10,14))
 			var/datum/effect_system/smoke_spread/smoke = new
-			smoke.set_up(3, 0, src.loc)
+			smoke.set_up(3, FALSE, src.loc)
 			smoke.attach(src)
 			smoke.start()
 			explosion(src.loc, -1, 0, 1, 3, 1, 0)
```
Comment (Contrabang) [inline code/modules/power/smes.dm:431]:
```suggestion
			smoke.set_up(3, FALSE, loc)
			smoke.attach(src)
			smoke.start()
			explosion(loc, -1, 0, 1, 3, 1, 0)
```

File: code/modules/power/smes.dm [hunk 13]
```diff
@@ -439,7 +439,7 @@
 				emp_act(2)
 		if(prob(5)) //smoke only
 			var/datum/effect_system/smoke_spread/smoke = new
-			smoke.set_up(3, 0, src.loc)
+			smoke.set_up(3, FALSE, src.loc)
```
Comment (Contrabang) [inline code/modules/power/smes.dm:442]:
```suggestion
			smoke.set_up(3, FALSE, loc)
```

**Review by Contrabang — CHANGES_REQUESTED** [review]
src.loc review, if any of these are out of scope, leave em be

File: code/modules/atmospherics/machinery/pipes/simple/pipe_simple.dm (hunk 9 above)
Comment (S34NW) [reply code/modules/atmospherics/machinery/pipes/simple/pipe_simple.dm:92]: out of scope

File: code/game/objects/items/weapons/grenades/smokebomb.dm (hunk 7 above)
Comment (S34NW) [reply code/game/objects/items/weapons/grenades/smokebomb.dm:21]: out of scope

File: code/modules/mob/living/simple_animal/hostile/hivebot.dm (hunk 11 above)
Comment (S34NW) [reply code/modules/mob/living/simple_animal/hostile/hivebot.dm]: partially out of scope

File: code/modules/projectiles/projectile/magic.dm [hunk 14]
```diff
@@ -116,7 +116,7 @@
 			teleammount++
 			do_teleport(stuff, stuff, 10)
 			var/datum/effect_system/smoke_spread/smoke = new
-			smoke.set_up(max(round(10 - teleammount),1), 0, stuff.loc) //Smoke drops off if a lot of stuff is moved for the sake of sanity
+			smoke.set_up(max(round(10 - teleammount),1), FALSE, stuff.loc) //Smoke drops off if a lot of stuff is moved for the sake of sanity
```
Comment (lewcc) [inline code/modules/projectiles/projectile/magic.dm:119]:
```suggestion
			smoke.set_up(max(round(10 - teleammount), 1), FALSE, stuff.loc) //Smoke drops off if a lot of stuff is moved for the sake of sanity
```

File: code/game/mecha/mecha.dm [hunk 15]
```diff
@@ -108,7 +108,7 @@
 	add_airtank()
 	spark_system.set_up(2, 0, src)
 	spark_system.attach(src)
-	smoke_system.set_up(3, src)
+	smoke_system.set_up(3, FALSE, src)
```
Comment (farie82) [inline code/game/mecha/mecha.dm:111]: So this one basically just didn't work before? Can you add this to the CL if that were the case?

File: code/modules/projectiles/projectile/magic.dm (hunk 14 above)
Comment (farie82) [reply code/modules/projectiles/projectile/magic.dm:119]:
```suggestion
			smoke.set_up(max(round(10 - teleammount), 1), FALSE, stuff) //Smoke drops off if a lot of stuff is moved for the sake of sanity
```
You mean like this @lewcc ?

**Review by farie82 — APPROVED** [review]

**Pull request merged**